### PR TITLE
fix(insights): show deprecation warnings for old insights related properties and functions

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR deprecates and shows warning messages for the following things:
- `getInsightsAnonymousUserToken()`
- `InstantSearch({ insightsClient })` ← `insightsClient` prop
- `analytics` widget

### CODE REVIEW

File: src/helpers/get-insights-anonymous-user-token.ts [hunk 1]
```diff
@@ -15,6 +17,19 @@ function getCookie(name: string): string | undefined {
   return undefined;
 }
 
-export default function getInsightsAnonymousUserToken(): string | undefined {
+export function getInsightsAnonymousUserTokenInternal(): string | undefined {
```
Comment (eunjae-lee) [inline src/helpers/get-insights-anonymous-user-token.ts:20]: This logic is used internally by `insights` middleware. So we keep this for us.

**Comment by francoischalifour** [comment]
Did we make sure that the Insights middleware covers all cases that the `analytics` widget covers?

File: src/widgets/analytics/analytics.ts [hunk 2]
```diff
@@ -76,6 +77,13 @@ const analytics: AnalyticsWidget = function analytics(widgetParams) {
     throw new Error(withUsage('The `pushFunction` option is required.'));
   }
 
+  warning(
+    false,
+    `\`analytics\` widget has been deprecated. It is still supported in 4.x releases, but not further. It is replaced by \`insights\` middleware.
```
Comment (Haroenv) [inline src/widgets/analytics/analytics.ts:82]: we should show here exactly how to replace it IMO, the analytics widget gets fired on different moments than the insights one I think. Maybe we can look at how magento uses it, and migrate there to the insights middleware before deprecating? Maybe adding a FAQ or a guide in "upgrade guides"?

Comment (eunjae-lee) [reply src/widgets/analytics/analytics.ts]: As soon as the doc PR is merged & deployed, I will add the link here.

**Review by Haroenv — APPROVED** [review]
looks good to me

**Pull request merged**